### ISSUE TITLE
Warning shown on email login field even when site has no Facebook tracking

### ISSUE THREAD

**Issue by kmahelona** [mozilla/contain-facebook#785]
- Facebook Container Version: Latest, which clearly explains what's happening,

> Added prompt for email fields to alert users about Facebook tracking by email address, and promote Firefox Relay. Note that there is a checkbox users can click when dismissing this prompt to no longer see the badge icon.

### Actual behavior
I'm currently being warned that Facebook may get my email for a number of websites with email logins, even our own website which has zero 3rd party cookies. I used to sometimes be warned about this, but now it seems like I'm being warned more often. Is there something more we need to do to ensure this warning doesn't show up for our website?

### Expected behavior
I'd only expect this warning to show when the site includes Facebook or other third party tracking. I could be totally wrong, but this kinda feels like a scare tactic especially when I'm encouraged to then sign up for the relay service. The warning insinuates that our site "can" share your email with Facebook. I mean sure, we could theoretically do anything but we [never](https://wharekōrero.nz/kaitiakitanga/) will.

This is totally unfair especially for small non-profits like ours that are [championing](https://papareo.nz/) [data sovereignty.](https://papareo.nz/#kaitiakitanga) For Mozilla to insinuate that we could give your email to Facebook is hurtful, and I'd expect much better from Mozilla.

Mahalo,
Keoni.

### Steps to reproduce
1. Go to https://wharekōrero.nz/iwi-login/
2. Inspect the code and network and see no communication with Facebook

### Notes
![Screen Shot 2021-08-03 at 12 42 51 PM](https://user-images.githubusercontent.com/5294466/127940436-2acb5d48-3730-460f-a62f-c0c235597e8f.png)
![Screen Shot 2021-08-03 at 12 42 58 PM](https://user-images.githubusercontent.com/5294466/127940440-56297cf5-07cf-4ab0-b579-caad63bfa504.png)

**Comment by grahamharris1975** [comment]
This also happens with a site that I am developing: 

https://churchrotadev.ddns.net/ 

![image](https://user-images.githubusercontent.com/13691012/127976910-edcc03fb-663f-49a2-be37-d1486f4f352d.png)
![image](https://user-images.githubusercontent.com/13691012/127977004-c8823bd7-d66d-4c9f-a3df-a678e7759eb5.png)

I have checked the network and storage tabs and there appears to be no facebook related content

**Comment by maxxcrawford** [comment]
This issue is a duplicate of #783. Note that we released a new version of the add-on to only show this message on websites were Facebook resources have been detected and blocked.

**Comment by lbruno** [comment]
i'll comment on this bug rather than the previous, to avoid triggering another pile-up. :)

Your users might need to refresh previously visited paged. My repro-case was:

1. visit https://www.paypal.com
2. go to the login page

Result: the badge is still there. I've just updated to 2.3.1, by going to `about:addons`, and paypal's login form still shows the badge.

Expected result: no badge

Fix: I hit Ctrl-R, and the badge disappeared and didn't come back. I verified this with another repro-case that I had on another website, which I knew wasn't cacheable so it didn't display this secondary problem after updating to 2.3.1

**Comment by maxxcrawford** [comment]
@lbruno – I'm testing this [locally on a fresh install](https://extensionworkshop.com/documentation/develop/getting-started-with-web-ext/#using-web-ext-section), and I'm **not** seeing the badge [1]. I can confirm there isn't any resources being blocked by FBC on paypal.com, so I agree – you shouldn't see any sort of badge on the email login field. 

URL wise, when I click the login button, I am directed to the US version of the login page: https://www.paypal.com/us/signin 
Is that the same URL you're checking against? 

However – if I click the "Sign Up" hero button, I'm directed to a new URL: https://www.paypal.com/us/welcome/flow/signup. That site does include blocked resources, so I do see the badge on that form (as expected) [2]. 

### Screentshots

[1] Sign in page: (`https://www.paypal.com/us/signin`)

<img width="600" alt="image" src="https://user-images.githubusercontent.com/2692333/128210043-dc711d4a-cdd4-4632-b9bb-40f58d52df93.png">

[2] "Sign Up" Hero: (`https://www.paypal.com/us/welcome/flow/signup`)

<img width="661" alt="image" src="https://user-images.githubusercontent.com/2692333/128209986-4bd29278-6117-4e54-a9dd-b38d5cf07e31.png">

**Comment by lbruno** [comment]
apologies for not being clear: i only saw this once, on the same page I had previously visited to check this out -- specifically, I had to refresh the cached page, then it never showed the badge again.

in my particular case it was the Ireland version (paypal.com/ie), but i don't think that's causing the issue here.

like, the caching of the page is the source of the issue: a fresh install most definitely won't show the badge, yes :) my point was that you should communicate a bit more proactively about the possible need to refresh -- problem is, I don't know if there's an appropriate channel for that proactive communication, other than the release notes themselves.

make it a note for the future -- i myself got surprised by the caching aspect, post-update of the extension.
